File: Hackerrank/30 Days of Code/Day25_Running_Time_and_Complexity.py

import math

number = input()

for i in range(int(number)):
    str_num = input()
    int_num = int(str_num)
    
    if(int_num==1):
        print("Not prime")
    else:
        k = 2
        for j in range(2, int(math.sqrt(int_num))+1):
            if(int_num%k==0):
                break
            k = k + 1

        if(k<=math.sqrt(int_num)):
            print("Not prime")
        else:
            print("Prime")
        

File: Hackerrank/30 Days of Code/Day25_Running_Time_and_Complexity.c
#include <stdio.h>
#include <string.h>
#include <math.h>
#include <stdlib.h>

int main() {

    int num = 0;

    scanf("%d", &num);
    for(int i=0; i<num; i++)
    {
        int input, j;
        scanf("%d", &input);
        
        if(input == 1)
            printf("Not prime\n");
        else
        {
            for(j=2; j<=sqrt(input); j++)
            {
                if(input%j==0)
                    break;      
            }

            if(j<=sqrt(input))
                printf("Not prime\n");
            else
                printf("Prime\n");
        }
    }
    return 0;
}
